app/controllers/MyProjects/DashboardController.php

```diff
@@ -119,6 +121,347 @@ function dashboard() {
  			$this->render('Dashboard/dashboard_html.php');
  		}
  		# -------------------------------------------------------
+ 		function dashboard() {
+ 			if(!$this->request->user->isFullAccessUser()){
+ 				$this->projectList();
+ 				return;
+ 			}
+ 			JavascriptLoadManager::register("cycle");
+			if(!$this->opn_project_id){
+				$this->projectList();
+				return;
+			}
+					
+			// Sort variable handling
+			$vs_specimens_group_by = 
```

Comment:
It's possible that a garbage getParameter value could result in different session and view variables getting set for the three parameters below